include/opc/common/interface.h

```diff
@@ -18,9 +18,15 @@ namespace Common
   {
   protected:
     Interface(){}
+
+    // do _not_ allow destructors to throw exceptions as it
+    // creates a compiler error on OSX and may break shared_ptr
+    // cleanup and cause memory leaks
+    /*
     // c++11 destructors default to noexcept
     // allow implementations to do so.
     virtual ~Interface() noexcept (false) {}
+    */
 
```

Comment:
As AddonsManager uses throw in its destructor but is not used in a shared_ptr context and one must not use shared_ptr instances with constructors or destructors which throw exceptions it may be a good idea to not derive AddonsManager from Interface - is this a possible solution?

On the other hand - throwing exceptions from destructors is discouraged in C++ anyway. So maybe AddonsManager should be modified to not do that.